bindings/python/pinocchio/visualize/viser_visualizer.py

```diff
@@ -0,0 +1,370 @@
+
+    def displayFrames(self, visibility):
+        self.framesRootFrame.visible = visibility
+        self.updateFrames()
+
+    def drawFrameVelocities(self, *args, **kwargs):
+        raise NotImplementedError("drawFrameVelocities is not yet implemented.")
+
+    def updatePlacements(self, geometry_type):
+        if geometry_type == pin.GeometryType.VISUAL:
+            geom_model = self.visual_model
+            geom_data = self.visual_data
+            prefix = self.viewerRootNodeName + "/visual"
+        else:
+            geom_model = self.collision_model
+            geom_data = self.collision_data
+            prefix = self.viewerRootNodeName + "/collision"
+
+        pin.updateGeometryPlacements(self.model, self.data, geom_model, geom_data)
+        for geom_id, geometry_object in enumerate(geom_model.geometryObjects):
+            # Get mesh pose.
+            M = geom_data.oMg[geom_id]
+
+            # Update viewer configuration.
+            frame_name = prefix + "/" + geometry_object.name
+            frame = self.frames[frame_name]
+            frame.position = M.translation * geometry_object.meshScale
+            frame.wxyz = pin.Quaternion(M.rotation).coeffs()[
+                [3, 0, 1, 2]
+            ]  # Pinocchio uses xyzw
+
+    def updateFrames(self):
+        pin.updateFramePlacements(self.model, self.data)
+        for frame_id, frame in enumerate(self.model.frames):
+            # Get frame pose.
+            M = self.data.oMf[frame_id]
+
+            # Update viewer configuration.
+            viser_frame_name = self.framesRootNodeName + "/" + frame.name
+            viser_frame = self.frames[viser_frame_name]
+            viser_frame.position = M.translation
+            viser_frame.wxyz = pin.Quaternion(M.rotation).coeffs()[
+                [3, 0, 1, 2]
+            ]  # Pinocchio uses xyzw
+
+    def captureImage(self, w=None, h=None, client_id=None, transport_format="jpeg"):
+        """
+        Capture an image from the Viser viewer and return an RGB array.
+
+        Parameters:
+            w: The width of the captured image.
+            h: The height of the captured image.
+            client_id: The ID of the Viser client handle.
+                If None, uses the first available client.
+            transport_format: The transport format to use for the captured image.
+                Can be "jpeg" (default) or "png".
+        """
+        clients = self.viewer.get_clients()
+        if len(clients) == 0:
+            raise RuntimeError("Viser server has no attached clients!")
+
+        if client_id is None:
+            cli = next(iter(clients.values()))
+        elif client_id not in clients:
+            raise RuntimeError(
+                f"Viser server does not have a client with ID '{client_id}'"
+            )
+        else:
+            cli = clients[client_id]
+
+        return cli.get_render(height=h, width=w, transport_format=transport_format)
```

Comment:
@sea-bass on the serialization error: `get_render()` expect integers for height and width, but it looks like this can end up passing `height=None` and `width=None`.

https://viser.studio/versions/1.0.0/api/handles/client_handles/#viser.ClientHandle.get_render

Reply:
That would do it, thank you!

I guess should we make these required arguments? Default them to integers?